runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1348,21 +1770,15 @@ TR::Register *J9::Power::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::Cod
 
    cg->machine()->setLinkRegisterKilled(true);
 
-   return objReg;
-#else
-   return NULL;
+   return objReg;   
 #endif
    }
 
-TR::Register *J9::Power::TreeEvaluator::ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return TR::TreeEvaluator::ardbariEvaluator(node, cg);
-   }
-
-TR::Register *J9::Power::TreeEvaluator::ardbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+TR::Register *aGenerateSoftwareReadBarrier(TR::Node *node, TR::CodeGenerator *cg)
    {
-   TR_ASSERT(TR::Compiler->om.readBarrierType() != gc_modron_readbar_none, "aReadBarrierEvaluator");
-#ifdef OMR_GC_CONCURRENT_SCAVENGER
+#ifndef OMR_GC_CONCURRENT_SCAVENGER
+   TR_ASSERT_FATAL(false, "Concurrent Scavenger not supported.");
```

Comment:
I think you still need a return statement after the assert fatal.

Reply:
TR_ASSERT_FATAL calls exit() which does not return. So a return would be unreachable code.